Dagger2 - Failed test in some build environment

After the change from `ServicesContext` to `Dagger2`, some build environment failed with the following errors:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.147 s <<< FAILURE! - in com.hedera.services.ServicesAppTest
[ERROR] com.hedera.services.ServicesAppTest.objectGraphRootsAreAvailable  Time elapsed: 2.137 s  <<< ERROR!
java.lang.IllegalStateException: Could not construct record stream manager
	at com.hedera.services.ServicesAppTest.objectGraphRootsAreAvailable(ServicesAppTest.java:123)
Caused by: java.nio.file.AccessDeniedException: /opt/hgcapp
	at com.hedera.services.ServicesAppTest.objectGraphRootsAreAvailable(ServicesAppTest.java:123)
```

This is because `RecordStreamManager` by default will use `hedera.recordStream.logDir=/opt/hgcapp/recordStreams` property and `/opt/hgcapp` is not accessible in some build environment.

### Solution:
Change `ServicesAppTest` to initialize `RecordStreamManager` from a different (accessible) `hedera.recordStream.logDir`

--- a/hedera-node/src/test/java/com/hedera/services/ServicesAppTest.java
+++ b/hedera-node/src/test/java/com/hedera/services/ServicesAppTest.java
@@ -9,9 +9,9 @@ package com.hedera.services;
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
+ *
  *      http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
@@ -24,8 +24,10 @@ import com.hedera.services.context.CurrentPlatformStatus;
 import com.hedera.services.context.NodeInfo;
 import com.hedera.services.context.init.ServicesInitFlow;
 import com.hedera.services.context.properties.BootstrapProperties;
+import com.hedera.services.context.properties.ChainedSources;
 import com.hedera.services.context.properties.GlobalDynamicProperties;
 import com.hedera.services.context.properties.NodeLocalProperties;
+import com.hedera.services.context.properties.PropertySource;
 import com.hedera.services.grpc.GrpcStarter;
 import com.hedera.services.grpc.NettyGrpcServerManager;
 import com.hedera.services.ledger.accounts.BackingAccounts;
@@ -69,6 +71,7 @@ import static org.hamcrest.Matchers.instanceOf;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertSame;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.BDDMockito.given;
 
 @ExtendWith(MockitoExtension.class)
@@ -92,13 +95,16 @@ class ServicesAppTest {
 	private ServicesState initialState;
 	@Mock
 	private RecordsRunningHashLeaf runningHashLeaf;
+	@Mock
+	private PropertySource overridingProps;
 
 	private ServicesApp subject;
 
 	@BeforeEach
 	void setUp() {
-		// setup:
 		final var bootstrapProps = new BootstrapProperties();
+		final var props = new ChainedSources(overridingProps, bootstrapProps);
+		final var logDir = "hedera.recordStream.logDir";
 
 		given(address.getStake()).willReturn(123_456_789L);
 		given(addressBook.getAddress(selfId)).willReturn(address);
@@ -108,9 +114,12 @@ class ServicesAppTest {
 		given(runningHash.getHash()).willReturn(hash);
 		given(platform.getCryptography()).willReturn(cryptography);
 		given(platform.getSelfId()).willReturn(selfNodeId);
+		given(overridingProps.containsProperty(any())).willReturn(false);
+		given(overridingProps.containsProperty(logDir)).willReturn(true);
+		given(overridingProps.getProperty(logDir)).willReturn("data/recordStreams");
 
 		subject = DaggerServicesApp.builder()
-				.bootstrapProps(bootstrapProps)
+				.bootstrapProps(props)
 				.initialState(initialState)
 				.platform(platform)
 				.selfId(selfId)
@@ -119,7 +128,6 @@ class ServicesAppTest {
 
 	@Test
 	void objectGraphRootsAreAvailable() {
-		// expect:
 		assertThat(subject.logic(), instanceOf(StandardProcessLogic.class));
 		assertThat(subject.hashLogger(), instanceOf(HashLogger.class));
 		assertThat(subject.workingState(), instanceOf(StateAccessor.class));
@@ -131,7 +139,7 @@ class ServicesAppTest {
 		assertThat(subject.nodeLocalProperties(), instanceOf(NodeLocalProperties.class));
 		assertThat(subject.recordStreamManager(), instanceOf(RecordStreamManager.class));
 		assertThat(subject.globalDynamicProperties(), instanceOf(GlobalDynamicProperties.class));
-		// and:
+
 		assertThat(subject.grpc(), instanceOf(NettyGrpcServerManager.class));
 		assertThat(subject.platformStatus(), instanceOf(CurrentPlatformStatus.class));
 		assertThat(subject.accountsExporter(), instanceOf(ToStringAccountsExporter.class));
@@ -149,7 +157,7 @@ class ServicesAppTest {
 		assertThat(subject.reconnectListener(), instanceOf(ReconnectListener.class));
 		assertThat(subject.grpcStarter(), instanceOf(GrpcStarter.class));
 		assertThat(subject.updateHelper(), instanceOf(UpdateHelper.class));
-		// and:
+
 		assertSame(subject.nodeId(), selfNodeId);
 		assertSame(subject.pause(), SLEEPING_PAUSE);
 		assertTrue(subject.consoleOut().isEmpty());

